data/leonovo-ideapad-duet.tablet

```diff
@@ -0,0 +1,15 @@
+#This is for the Google Kukui Lenovo Ideapad Duet
+
+[Device]
+Name=Lenovo Ideapad Duet
+Class=ISDV4
+DeviceMatch=i2c:27C6:0E30
+Width=8.5
+Height=5.25
+IntegratedIn=Display;System
+#Styli=@generic-no-eraser
+Styli=0x812
```

Comment:
If you run `libinput record`, the `ABS_MISC` event tells you the tool id and thus the styli to pick. Generally if you support 0x812 that means you also support the various other pens in that group - though 0x812 is the *very old* intuos pen (15+ years old now).

So my guess is that it's either `@generic-no-eraser` or the `@isdv4-aes` group or we need another set of pens defined here.

Reply:
Actually, in the libwacom repository there's `./tools/show-stylus.py` which will tell you the tool ID in slightly more human-readable format.

Reply:
> Actually, in the libwacom repository there's `./tools/show-stylus.py` which will tell you the tool ID in slightly more human-readable format.

So this is my output:

./show-stylus.py
Using "hid-over-i2c 27C6:0E30 Stylus": /dev/input/event5
Using stylus file(s): /usr/share/libwacom/libwacom.stylus
Tool id 0x1 serial 0x0 in-proximity: False 
^CTerminating
Suggested line for .tablet file:
Styli=isdv4-aes> Actually, in the libwacom repository there's `./tools/show-stylus.py` which will tell you the tool ID in slightly more human-readable format.

So this is my output:

`./show-stylus.py
Using "hid-over-i2c 27C6:0E30 Stylus": /dev/input/event5
Using stylus file(s): /usr/share/libwacom/libwacom.stylus
Tool id 0x1 serial 0x0 in-proximity: False 
^CTerminating
Suggested line for .tablet file:
Styli=isdv4-aes`

But 0x1 has a button and a eraser button, which mine doesn't. I also couldn't find one that has.

But 0x1 has a button and a eraser button, which mine doesn't. I also couldn't find one that has.

Reply:
0x27C6 says it's a Goodix USB ID so basically what we have here is a stylus that's not a wacom stylus and libwacom implicitly assumes that all styli with IDs are wacoms.

So if the ID is "re-used" we need to fix libwacom to be able to tell the difference between wacom styli and non-wacom styli. Then we can add a new definition for this stylus with `72C6:1`. I wonder what the effect of this will be on the API...